netbox/dcim/tests/test_filtersets.py

```diff
@@ -4254,4 +4254,121 @@ def test_connected(self):
+            VirtualDeviceContext(device=devices[2], name='VDC 1', status=VirtualDeviceContextStatusChoices.STATUS_ACTIVE),
+            VirtualDeviceContext(device=devices[2], name='VDC 2', status=VirtualDeviceContextStatusChoices.STATUS_ACTIVE),
+        )
+        VirtualDeviceContext.objects.bulk_create(vdcs)
+
+        interfaces = (
+            Interface(device=devices[0], name='Interface 1', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[0], name='Interface 2', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[1], name='Interface 3', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[1], name='Interface 4', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[2], name='Interface 5', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[2], name='Interface 6', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[0], name='Interface 7', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[0], name='Interface 8', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[1], name='Interface 9', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[1], name='Interface 10', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[2], name='Interface 11', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+            Interface(device=devices[2], name='Interface 12', type=InterfaceTypeChoices.TYPE_1GE_FIXED),
+        )
+        Interface.objects.bulk_create(interfaces)
+
+        interfaces[0].vdcs.set([vdcs[0], vdcs[1]])
+        interfaces[1].vdcs.set([vdcs[0], vdcs[1]])
+        interfaces[2].vdcs.set([vdcs[2], vdcs[3]])
+        interfaces[3].vdcs.set([vdcs[2], vdcs[3]])
+        interfaces[4].vdcs.set([vdcs[4], vdcs[5]])
+        interfaces[5].vdcs.set([vdcs[4], vdcs[5]])
+        interfaces[6].vdcs.set([vdcs[0]])
+        interfaces[7].vdcs.set([vdcs[1]])
+        interfaces[8].vdcs.set([vdcs[2]])
+        interfaces[9].vdcs.set([vdcs[3]])
+        interfaces[10].vdcs.set([vdcs[4]])
+        interfaces[11].vdcs.set([vdcs[5]])
+
+
+class VirtualDeviceContextTestCase(VirtualDeviceContextBaseTestCase, TestCase, ChangeLoggedFilterSetTests):
+    queryset = VirtualDeviceContext.objects.all()
+    filterset = VirtualDeviceContextFilterSet
+
+    def test_device(self):
+        params = {'device': ['Device 1', 'Device 2']}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 6)
+
+    def test_status(self):
+        params = {'status': ['active']}
+        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 3)
+
+
+class VirtualDeviceContextInterfaceTestCase(VirtualDeviceContextBaseTestCase, TestCase, ChangeLoggedFilterSetTests):
```

Comment:
I'm not sure about adding these here. IMO every test for interface filters should go under the `InterfaceTestCase` class. I realize the advantage of making a separate class here is to reuse the object creation code, however I worry that this will needlessly complicate potential future evaluations. Also, it should be fairly simple to extend InterfaceTestCase with some VDC assignments.

Reply:
Good point, I wrapped it into the InterfaceTestCase.